### PR TITLE
Cleanup filesystem

### PR DESCRIPTION
**Pull request by DanRStevens**
A bit of cleanup to support work on #967.

### CODE REVIEW

File: NAS2D/Filesystem.cpp [hunk 1]
```diff
@@ -35,6 +35,12 @@ namespace {
 	using SdlString = std::unique_ptr<char, SdlStringDeleter>;
 
 
+	bool hasFileSuffix(const std::string& filePath, const std::string& suffix)
+	{
+		return filePath.rfind(suffix, filePath.length() - suffix.length()) != std::string::npos;
```
[the remark this answers is not in the document]
Comment (DanRStevens) [reply NAS2D/Filesystem.cpp:40]: Noted. I would definitely like to see improvements here at some point. I assume we'll be making the C++20 shift soon. Alternatively, we could start using `std::filesystem::path` to extract the suffix, though admittedly that doesn't work quite the same as this code (`t` is a suffix of `.txt`, though isn't the full extension).

Related: #826, #1019, #973

**Comment by DanRStevens** [comment]
Not sure I entirely understand what Codacy is complaining about here:
https://app.codacy.com/gh/lairworks/nas2d-core/pullRequest?prid=9772250
> Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).

There are no loops, nor recursion in the called method. I wonder if perhaps the error relates to expanding a Google Test macro.

The Codacy page says this flaw was reported by `flawfinder`, which I assume they're using a plugin for:
https://dwheeler.com/flawfinder/

I may look into this when I have a bit more time.

File: NAS2D/Filesystem.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (cugone) [reply NAS2D/Filesystem.cpp:40]: Just for clarification, when you mean "suffix" do you mean ["extension"](https://en.cppreference.com/w/cpp/filesystem/path/extension), ["stem"](https://en.cppreference.com/w/cpp/filesystem/path/stem), or ["filename"](https://en.cppreference.com/w/cpp/filesystem/path/filename)? e.g. given the path `Foo/Bar/Baz.txt`: the extension is `.txt`, the filename is `Baz.txt`, and the stem is `Baz`. (Depending on the location of the last period, these have different results as well. See the accompanying links.)

[the remark this answers is not in the document]
Comment (DanRStevens) [reply NAS2D/Filesystem.cpp:40]: Well that's my point. The "suffix" code above is roughly something like "extension", but it's not actually the same. The suffix doesn't care about any ".".

**Comment by DanRStevens** [comment]
PR #1026 fixed the above mentioned Codacy errors (including similar errors affecting `main`). I've rebased this work on top of those changes.

**Pull request merged**